pebblo/app/service/doc_helper.py

```diff
@@ -41,49 +146,223 @@ def _get_classifier_response(self, doc):
             logger.error(f"Get Classifier Response Failed, Exception: {e}")
             return doc_info
 
-    def _get_finding_details(self, doc, data_source_findings, entity_type, file_count, raw_data):
+    def _update_app_details(self, raw_data, ai_app_docs):
+        """
+            Updating ai app details loader source files
+        """
+        logger.debug("Updating app details")
+        self.app_details["docs"] = ai_app_docs
+        loader_source_snippets = raw_data["loader_source_snippets"]
+        # Updating app_details doc list and loader source files
+        loader_details = self.app_details.get("loaders", {})
+        for loader in loader_details:
+            for source_file in loader.get("sourceFiles", []):
+                name = source_file["name"]
+                if name not in loader_source_snippets:
+                    loader_source_snippets[name] = source_file
+
+            new_source_file = [{
```

Comment:
nitpick: `new_source_file` is named in the singular form, which might be misleading since it contains a list